### ISSUE TITLE
Can't insert to table storage from async void method

### ISSUE THREAD

**Issue by GuardRex** [Azure/azure-storage-net#169]
I thought that an async method returning void would be able to perform a table insert, but I'm having an issue with it.

If I return a Task&lt;T&gt;, this works ...
```c#
public static async Task<string> LogError(string errorMessage) {
    ErrorLog entity = new ErrorLog((DateTime.MaxValue - DateTime.UtcNow).ToString());
    entity.Error = errorMessage;
    StorageCredentials storageCredentials = new StorageCredentials("name", "key");
    CloudStorageAccount storageAccount = new CloudStorageAccount(storageCredentials, true);
    CloudTableClient tableClient = storageAccount.CreateCloudTableClient();
    tableClient.DefaultRequestOptions.RetryPolicy = new ExponentialRetry(TimeSpan.FromSeconds(1), 10);
    CloudTable table = tableClient.GetTableReference("ErrorLog");
    await table.ExecuteAsync(TableOperation.Insert(entity));
    return "done";
}
```
... but it fails if I set it up to return void ...
```c#
public static async void LogError(string errorMessage) {
    ErrorLog entity = new ErrorLog((DateTime.MaxValue - DateTime.UtcNow).ToString());
    entity.Error = errorMessage;
    StorageCredentials storageCredentials = new StorageCredentials("name", "key");
    CloudStorageAccount storageAccount = new CloudStorageAccount(storageCredentials, true);
    CloudTableClient tableClient = storageAccount.CreateCloudTableClient();
    tableClient.DefaultRequestOptions.RetryPolicy = new ExponentialRetry(TimeSpan.FromSeconds(1), 10);
    CloudTable table = tableClient.GetTableReference("ErrorLog");
    await table.ExecuteAsync(TableOperation.Insert(entity));
}
```
... it outputs ...
```
Exception thrown: 'System.InvalidOperationException' in mscorlib.ni.dll
```
... on the `.ExecuteAsync` line. The app is:
```
dnx-coreclr-win-x64.1.0.0-beta6
Web API using the MVC.Core package
IIS Express, VS 2015 RTM, Win10 RTM
```
Dependencies:
```
"Microsoft.AspNet.Mvc.Core": "6.0.0-beta6",
"Microsoft.AspNet.Server.IIS": "1.0.0-beta6",
"System.Runtime.Serialization.Primitives": "4.0.10-beta-*",
"Microsoft.AspNet.Mvc.Formatters.Json": "6.0.0-beta6",
"WindowsAzure.Storage": "5.0.1-preview",
"Microsoft.Framework.Configuration": "1.0.0-beta6",
"xunit.runner.aspnet": "2.0.0-aspnet-beta6-*"
```

**Comment by Cheesebaron** [comment]
Do you have a stack trace on that InvalidOperationException?

**Comment by GuardRex** [comment]
Ah! I got it. Rubber duck to the rescue.

The problem wasn't that the table insert method was setup to return void ... it was that I had the `HttpPut` setup to return void in the controller. It sure didn't care for that!

I changed the controller to return a `Task<string>`. I set my table insert method back to return void. After calling that method in the controller, I followed it with a `Context.HttpContext.Response.StatusCode = 201;` and `return string.Empty;`. It stopped choking. All is well.

The exception, as with many exceptions in .NET 5, is really, really unhelpful ... but that's _**beta**_ for ya.
